### PR TITLE
Edit/delete emails and validate inputs

### PR DESCRIPTION
**Pull request by chukwumaijem**
#### What does this PR do?
- Validate inputs
- Enable email editing and deleting in email settings

### CODE REVIEW

File: client/src/components/Settings/SingleEmail.js [hunk 1]
```diff
@@ -34,6 +59,15 @@ class SingleEmail extends React.Component {
     this.toggleDisable();
   }
 
+  removeEmail() {
+    const email = this.props.email;
+    const content = {
+      userId: email.userId,
+      id: email.id
+    };
+    this.props.deleteEmail(content);
```
Comment (emjaynotmj) [inline client/src/components/Settings/SingleEmail.js:68]: Can we have `Toastr` message added to this action?

File: client/src/components/Settings/SingleEmail.js [hunk 2]
```diff
@@ -8,14 +12,35 @@ class SingleEmail extends React.Component {
...
-    const email = e.target.value;
-    this.setState({ email });
+
+  updateEmail() {
+    const newEmail = this.refs.email.value;
+    const email = this.props.email;
+    const regex = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+
+    if(email.email === newEmail) {
+      return Toastr.info('No change was made to this email: ' + newEmail);
```
Comment (emjaynotmj) [inline client/src/components/Settings/SingleEmail.js]: I think its better to use `No changes were made`

Comment (chukwumaijem) [reply client/src/components/Settings/SingleEmail.js]: Sure.

File: client/src/components/Settings/SingleEmail.js (hunk 1 above)
Comment (chukwumaijem) [reply client/src/components/Settings/SingleEmail.js:68]: Okay

**Pull request merged**